File: 023-digits-letters/digitsOrLetters.js
/*
Pseudocode
1. Use regex to count no of digits and no of letters, store values in var
2. Filter letters/numbers into an arr
3. Compare length of both arrays, if digits/letters are more or both are equal and return appropriate ans
*/

function digitsOrLetters(str) {
  const digReg = /[0-9]/;
  const letReg = /[a-z]/i;
  const arr = str.split("");
  const letters = arr.filter(i => letReg.test(i));
  const nums = arr.filter(i => digReg.test(i));
  let answer = letters.length > nums.length ? "letters" : nums.length > letters.length ? "digits" : "tie"
  console.log(answer);
}

File: 023-digits-letters/digits_or_letters.py
import re

'''
Pseudocode
1. Use regex to check for digits/letters,
2. Store all letters/digits found in str in list arr
3. Compare length of both lists, if digits/letters are more or both are equal and return appropriate ans
'''

def digits_or_letters(text):
  dig_reg = r"[0-9]"
  let_reg = r"[a-zA-Z]"
  letters = re.findall(let_reg, text)
  nums = re.findall(dig_reg, text)
  answer = "letters" if len(letters) > len(nums) else "digits" if len(nums) > len(letters) else "tie"
  print(answer)